src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -548,64 +549,66 @@ def lfnsInBlock(self, fileBlockName):
             raise DBSReaderError(msg)
 
 
-    def listFileBlockLocation(self, fileBlockName, dbsOnly = False, phedexNodes=False):
+    def listFileBlockLocation(self, fileBlockNames, dbsOnly = False):
         """
         _listFileBlockLocation_
 
         Get origin_site_name of a block
 
         """
-        blockNames = [fileBlockName] if isinstance(fileBlockName, basestring) else fileBlockName
-        for block in blockNames:
+        
+        singleBlockName = None
+        if isinstance(fileBlockNames, basestring):
+            singleBlockName = fileBlockNames
+            fileBlockNames = [fileBlockNames]
+ 
+        for block in fileBlockNames:
             self.checkBlockName(block)
 
-        blockInfo = {}
+        locations = {}
+        node_filter = set(['UNKNOWN', None])
         if not dbsOnly:
             try:
-                blockInfo = self.phedex.getReplicaSEForBlocks(phedexNodes=phedexNodes, block=blockNames, complete='y')
-            except Exception as ex:
+                blocksInfo = self.phedex.getReplicaPhEDExNodesForBlocks(block=fileBlockNames, complete='y')
+            except Exception, ex:
```

Comment:
oops yes this has come from rebasing my changes onto the head looks like it's a change I've introduced. I'll ensure that this is changed back before this branch is pulled.